content/en/blog/2025/expose-otel-collector-gateway-api/index.md

```diff
@@ -0,0 +1,619 @@
+---
+title: Exposing OTel Collector in Kubernetes with Gateway API & mTLS
+linkTitle: Collector with Gateway API and mTLS
+date: 2025-04-06
+author: >
+  [Vipin Vijaykumar](https://github.com/vipinvkmenon)
+sig: End-User SIG
+cSpell:ignore: gateway gatewayclass ingress ingressgateway Vijaykumar Vipin
+---
+
+The goal of this blog post is to demonstrate how you can expose an OpenTelemetry
+(OTel) Collector running inside Kubernetes to the outside world securely, using
+the [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/) and
+[mutual TLS (mTLS)](https://www.buoyant.io/mtls-guide) for authentication and
+encryption.
+
+As observability becomes increasingly critical in modern distributed systems,
+centralizing telemetry data via OTel Collectors deployed in one or many
+Kubernetes clusters is common practice. Often, services or agents running
+_outside_ your Kubernetes cluster need to send data _to_ these Collectors.
+Exposing internal services requires careful consideration of security and
+standardization. This is where the Kubernetes Gateway API and mTLS shine.
+
+Typically this kind of a setup is useful when you have applications or workloads
+that are external to the cluster, and you need to collect their telemetry data.
+Some examples:
+
+- **Hybrid Cloud/On-Premises Environments:** Applications or servers running in
+  a traditional data center, or a different cloud or external to your Kubernetes
+  cluster need to forward their metrics, traces, or logs to your central
+  observability solution.
+- **Multi-Cluster Telemetry Aggregation:** In a setup that might run on multiple
+  Kubernetes clusters, you might designate one cluster to host the primary OTel
+  Collector deployment. Collectors in other "spoke" clusters would act as
+  clients, exporting data to this central collector via its external endpoint.
+  For example, in a multi-cluster service mesh setup, workloads may require tail
+  sampling to be performed centrally. In this case, a central collector
+  configured with the
+  [tail sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor)
```

Comment:
Pin the URL to the most current version.

Reply:
Pinning it to the latest `v0.127.0`